README.md

```diff
@@ -1,4 +1,10 @@
-# UCAN Invocation Specification v1.0.0-rc.1
+# UCAN Invocation Specification v1.0.0-rc. 1
+
+FIXME
+- move pipelines, await, etc to own spec
+- explain in FAQ why we don't need `join` (i.e. use promises, since the model is inverted)
```

Comment:
Ah, nice! Could `fork` be expressed with promises as well?
Wondered about this idea of expressing effects via awaits [down below](https://github.com/ucan-wg/invocation/pull/21/commits/ee652ef705c2050820fc4405e894499d0ed83732#r1345687658).

Reply:
No, this is communication back to the Invoker, so it's not signed. We can subsume `join` by what used to be called `fork`, though.

Reply:
> Could fork be expressed with promises as well?
>> No, this is communication back to the Invoker, so it's not signed.

Aha, I see, thanks for clarifying!

> Enqueued [Task](https://github.com/ucan-wg/invocation/tree/high-level#32-task)s describe requests for future work to be performed.

These Tasks seem to be an interpretation of the Result.
E.g., given Result for a query returned 100 solutions => send an email.
I.e., enqued Tasks are some further actions based on / derived from the Result.

This interpretation / `deriveFurtherActions(Result)` could happen as part of the original Task, and it seems possible to perform it outside of it as well. Would it be valuable to have it outside, decoupling a Task from interpretation of its Result?